src/server/queries/suppliers.js

```diff
@@ -39,9 +39,22 @@ module.exports.exists = function(supplierId)
   return this.db.models.Supplier.findById(supplierId).then(supplier => supplier && supplier.supplierId === supplierId);
 }
 
+module.exports.recordExists = function(supplier)
+{
+  var options = { supplierName: supplier.supplierName };
+
+  var fields = ['registrationNumber', 'taxId', 'vatRegNo'];
+
+  for (var i = 0; i < fields.length; i++) {
```

Comment:
I used `for` here because of this https://coderwall.com/p/kvzbpa/don-t-use-array-foreach-use-for-instead :)